test/onnx/test_pytorch_onnx_onnxruntime.py

```diff
@@ -9440,6 +9442,45 @@ def forward(self, x):
         x = torch.randn(10, 5)
         self.run_test(M(), (x,))
 
+    def test_onnx_checker_invalid_graph(self):
+        op_source = """
+        #include <torch/script.h>
+
+        torch::Tensor custom_add(torch::Tensor self, torch::Tensor other) {
+          return self + other;
+        }
+
+        static auto registry =
+          torch::RegisterOperators("custom_namespace::custom_add", &custom_add);
+        """
+
+        torch.utils.cpp_extension.load_inline(
+            name="custom_add",
+            cpp_sources=op_source,
+            is_python_module=False,
+            verbose=True,
+        )
+
+        class CustomAddModule(torch.nn.Module):
+            def forward(self, a, b):
+                return torch.ops.custom_namespace.custom_add(a, b)
+
+        def invalid_symbolic(g, self, other):
+            return g.op("Add", self, other, invalid_attr_i=1)
+
+        register_custom_op_symbolic("custom_namespace::custom_add", invalid_symbolic, 9)
+
+        x = torch.randn(2, 3, 4)
+        y = torch.randn(2, 3, 4)
+
+        test_model = CustomAddModule()
+        f = io.BytesIO()
+
+        with self.assertRaises(RuntimeError) as cm:
+            torch.onnx.export(test_model, (x, y), f)
+
+        self.assertTrue(f.getvalue(), "ONNX graph was not generated")
```

Comment:
Can we try loading the model using onnx.load_model? If so I think that would be a more thorough test.

Reply:
I agree that if we add one step to verify the loading will make it a more thorough test.

But my understanding is: this is something like Unit Test, not function test. We should make each test small, focused and reliable.

For this test, actually we want to verify those new code which generates ONNX graph even it is invalid. As long as the ONNX graph has been generated, then our code has been verified.

If we added onnx.load_model() method here, actually we were also trying to verify that onnx_checker could identify invalid ONNX graph as expected. I think this scenario should not be covered by a test case for torch.onnx.export() function.

Reply:
I think we should just verify that the output is a complete onnx graph proto, which I think we can do by running `onnx.load_model`. `onnx.load_model` does not call the ONNX checker.

If we output some bytes that are not loadable as an ONNX model, then users will not be able to easily use it to debug.